resources/integration_test/testdata/calcs.yml

```diff
@@ -2,458 +2,576 @@ dataset:
 - collection: calcs
   db: tdvt
   docs:
-  - bool0: '1'
-    bool1: '1'
-    bool2: '0'
-    bool3: '1'
-    date0: '2004-04-15'
-    date1: '2004-04-01'
-    date2: '1977-04-20'
-    date3: '1986-03-20'
-    datetime0: '2004-07-09 10:17:35'
-    datetime1: ''
-    int0: '1'
-    int1: '-3'
-    int2: '5'
-    int3: '8'
+  - bool0: true
+    bool1: true
+    bool2: false
+    bool3: true
+    date0:
+      $date: '2004-04-15T00:00:00Z'
+    date1:
+      $date: '2004-04-01T00:00:00Z'
+    date2:
+      $date: '1977-04-20T00:00:00Z'
+    date3:
+      $date: '1986-03-20T00:00:00Z'
+    datetime0:
+      $date: '2004-07-09T10:17:35Z'
+    datetime1: null
+    int0:
+      $numberInt: 1
+    int1:
+      $numberInt: -3
+    int2:
+      $numberInt: 5
+    int3:
+      $numberInt: 8
     key: key00
-    num0: '12.3'
-    num1: '8.42'
-    num2: '17.86'
-    num3: '-11.52'
-    num4: ''
+    num0: 12.3
+    num1: 8.42
+    num2: 17.86
+    num3: -11.52
+    num4: null
     str0: FURNITURE
     str1: CLAMP ON LAMPS
     str2: one
     str3: e
     time0: '1899-12-30 21:07:32'
```

Comment:
I decided not to convert the time types since we don't actually support time and they are mixed dates and datetimes. I think it's fine to leave them as strings.

Reply:
I tried this in Power BI and it was able to transform the strings to date, datetime, and time by using `Change Type` on the column.  So it is fine to have here as strings.